### PR TITLE
Fix race conditions

### PR DESCRIPTION
**Pull request by lgrahl**
Fix a bunch of race conditions.

### CODE REVIEW

File: src/services/webclient.ts [hunk 1]
```diff
@@ -445,14 +447,17 @@ export class WebClientService {
      *          messages can be queued by the user.
      */
     public init(flags: {
-        keyStore?: saltyrtc.KeyStore,
-        peerTrustedKey?: Uint8Array,
+        keyStore?: saltyrtc.KeyStore | 'reuse',
+        peerTrustedKey?: Uint8Array | 'reuse',
```
Comment (dbrgn) [inline src/services/webclient.ts]: Can you document the 'reuse' arguments in the docstring?

**Comment by lgrahl** [comment]
I'm delaying this for now since I've seen an infinite loop regarding WebSocket connection buildup and teardown.

**Comment by dbrgn** [comment]
Reopening.

A problem we're currently having (since 2.2.0): When using an iOS device, on disconnecting (just lock the screen), we get this exception:

```
angular.js?v=2.2.0:15567 TypeError: Cannot read property 'permanentSharedKey' of null
    at InitiatorSignaling.get$$1 (saltyrtc-client.es5.js?v=2.2.0:1984)
    at SaltyRTC.get$$1 (saltyrtc-client.es5.js?v=2.2.0:3293)
    at e.value (status.ts:107)
    at e.value (status.ts:46)
    at status.ts:17
    at Scope.$eval (angular.js?v=2.2.0:19393)
    at Scope.$apply (angular.js?v=2.2.0:19492)
    at r.<anonymous> (status.ts:16)
    at r.t._call (base-event.js:147)
    at r._call (async-event.js:116)
```

**Comment by lgrahl** [comment]
I might have been able to close the demon portal. Let's test this carefully @dbrgn...

**Comment by dbrgn** [comment]
I haven't noticed any major problems with this branch during the last 10 days.

**Pull request merged**